File: SiriusJavaClient-Web/src/main/java/org/sirius/client/web/classes/Page.java
/**
 * 
 */
package org.sirius.client.web.classes;

import java.rmi.RemoteException;

import org.sirius.client.web.WebClient;

/**
 * @author Myk Kolisnyk
 * 
 */
public class Page extends Frame {

    public Page(final WebClient client) {
        super(client, null);
    }

    /**
	 * 
	 */
    public Page(final WebClient client, final String locator) {
        super(client, locator);
    }

    public void back() throws RemoteException {
        client.core().back();
    }

    public void forward() throws RemoteException {
        client.core().forward();
    }

    public void home() {
        ;
    }

    public void open(final String URL) throws RemoteException {
        client.core().open(URL);
    }

    public void refresh() throws RemoteException {
        client.core().refresh();
    }

    public void switchTo(final Page page) throws RemoteException {
        client.core().selectWindow(page.getLocator());
    }

    public String title() throws RemoteException {
        return client.core().getTitle();
    }
}

File: SiriusCSharp.Client/Sirius.CSharp.Client.Web/Classes/Page.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Sirius.CSharp.Client.Web.Classes
{
    public class Page : Frame
    {
        public Page(WebClient client, String locator)
            : base(client, locator)
        {
            ;
        }

        public Page(WebClient client)
            : base(client, null)
        {
            ;
        }

        public String title()
        {
            return client.getTitle();
        }

        public void back()
        {
            client.back();
        }

        public void forward()
        {
            client.forward();
        }

        public void refresh()
        {
            client.refresh();
        }

        public void open(String URL)
        {
            client.open(URL);
        }

        public void home()
        {
            ;
        }

        public void switchTo(Page page)
        {
            client.selectWindow(page.getLocator());
        }
    }
}
